Update Pedigree class (pedigree_v2) and test files.

stevenhwu opened the pull request:
This will be a drop-in replace of the existing class.

The update version will allow more detailed testings, and might be easier to handle different peeling operations if required.
Currently, this class (pedigree_v2) is not used in call.cc yet.

test/CMakeLists.txt might cause conflict later, because not all library are included.
test/src/fixture/ include fixture for boot_test that shared between multiple tests.

stevenhwu committed:
Update Pedigree class (pedigree_v2) and test files. This will be a drop-in replace of the existing class.
The update version will allow more detailed testings, and might be easier to handle different peeling operations if required.
Currently, this class (pedigree_v2) is not used in call.cc yet.

test/CMakeLists.txt might cause conflict later, because not all library are included.
test/src/fixture/ include fixture for boot_test that shared between multiple tests.

dng-jenkins commented:
Can one of the admins verify this patch?

reedacartwright commented:
Jenkins, ok to test.

reedacartwright commented on src/include/utils/boost_utils.h:
```diff
@@ -0,0 +1,23 @@
+//
+// Created by steven on 2/4/16.
+//
+
```
Missing boost headers

stevenhwu replied:
Updated at the next commit.
Both this and call.cc can compile and run without any #include <boost/**>. Is this a bad sign?

reedacartwright commented on src/include/utils/assert_utils.h:
```diff
@@ -0,0 +1,70 @@
+#define DENOVOGEAR_ASSERT_UTILS_H
+
+#include <cassert>
+
```
Please wrap these functions such that if NDEBUG is defined they evaluate to no-ops.

reedacartwright commented on src/include/utils/assert_utils.h:
```diff
@@ -0,0 +1,70 @@
+void AssertEqual(A expected, B actual) {
+    assert(expected == actual);
+};
+
```
Also, all these function names need to be in lower_case: e.g. `assert_equal`.

reedacartwright commented:
It is hard to see what is being changed since you put the new code in different files.

stevenhwu committed:
Addressed the following
- Missing boost headers
- Please wrap these functions such that if NDEBUG is defined they evaluate to no-ops.
- Also, all these function names need to be in lower_case: e.g. assert_equal.
  -- only a few files atm. At some stage, there will to a "code conventions" PR, which will  address other issues like this.

stevenhwu committed:
Addressed the following
- Missing boost headers
- Please wrap these functions such that if NDEBUG is defined they evaluate to no-ops.
- Also, all these function names need to be in lower_case: e.g. assert_equal.
  -- only a few files atm. At some stage, there will to a "code conventions" PR, which will  address other issues like this.

stevenhwu commented:
Update some of these comments.
Let's discuss about "It is hard to see what is being changed since you put the new code in different files." later.

stevenhwu committed:
Merge branch 'feature_pedigree_v2' of github.com:stevenhwu/denovogear into feature_pedigree_v2

dng-jenkins commented:
Can one of the admins verify this patch?

stevenhwu commented:
Can jenkin ignore this PR please.

dng-jenkins commented:
Can one of the admins verify this patch?
